### DISCUSSION TITLE
What approach to use to test models?

### DISCUSSION THREAD

**Discussion by dkrikun** [textX/textX#332] [Q&A]
I have set up a grammar, metamodel etc. Now, I would like to test the parser by feeding it various source codes as input and inspecting the resulting models.

At first, I thought, I would construct an expected model by hand and then compare it with the actual one (produced from parsing). However, I am not sure I can achieve this without defining everything with custom classes. Is it possible?

In general, what would be an idiomatic way to do such testing?

**Comment by igordejanovic** [comment]
We haven't done that before. A usual approach is to construct the model using textX and than to make some interesting assertion about the model (depending on what we are testing). No full comparison is done.

**Comment by igordejanovic** [comment]
Just a thought. Maybe something along this lines #38 could be used to provide full comparison.

**Comment by dkrikun** [comment]
Actually, I am now convinced it is more useful to traverse the model and assert various facts about it, because if "it walks like a duck and it quacks like a duck, then it must be a duck". In other words, full comparison is not necessary (IMHO)
